---
title: "Analyze initiation codon pair mRNA levels"
author: "`r Sys.info()[['user']]`"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: 
  github_document:
    toc: 2
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=F, message=F)
```


## Load libraries and define analysis-specific parameters
```{r}
# standard analysis and plotting functions, includes dplyr, ggplot2 
library(tidyverse)
# loads lab default ggplot2 theme and provides color-blind friendly palette
library(rasilabRtemplates)

codonmutation_order <- seq(1,4)
names(codonmutation_order) <- c('AAG', 'AGA', 'CCG', 'CCA')
# intiiation sites are arranged in this order
initiationmutation_order <- seq(1,9)
names(initiationmutation_order) <- toupper(c( 'ctg', 'ctgc', 'ccgc', 
                              'acgc', 'ccga', 'ccac', 'ccaa', 'caaa', 'aaaa'))
```

## Read annotations
```{r}
annotations <- read_tsv("../../../data/htseq/initiation_pgk1_aag_ccg/sampleannotations.tsv") %>% 
  print()
```

## Read count data
```{r}
count_data <- list.files("../../../tables/htseq/initiation_pgk1_aag_ccg/", 
                         pattern = "init_codon_pairs.tsv",
                         full.names = T) %>% 
  enframe("sno", "file") %>% 
  mutate(barcode = str_extract(file, "(?<=/)[ACTG]{6}")) %>% 
  mutate(data = map(file, read_tsv)) %>% 
  unnest() %>% 
  select(-sno, -file, -index) %>% 
  left_join(annotations %>% select(-primer), by = "barcode") %>% 
  filter(count > 100) %>% 
  arrange(desc(count)) %>% 
  print()
```

## Analyze PCR template switching between initiation mutation and codon mutation 

The wild-type PGK1 gene has a serine codon at the 4th position 5' to the end
of PGK1. 
These were cloned with only the CAAA Kozak sequence. 
This should not occur with any other initiation mutation.
By looking at the frequency of these codons with non-CAAA initiation region,
we can get an estimate of template-switching frequency.

It looks like at most 12% of reads come from template switching.
```{r}
count_data %>% 
  filter(type == "cdna" & replicate == 2) %>% 
  mutate(codon = substr(codon, 1, 3)) %>% 
  filter(codon %in% c("AGC", "AGT", "TCA", "TCC", "TCG", "TCT")) %>% 
  mutate(init = if_else(str_detect(init, "^CAAA"), "correct", "wrong")) %>% 
  group_by(codon, init) %>% 
  summarize(fraction = sum(count)) %>% 
  ungroup() %>% 
  group_by(codon) %>% 
  mutate(fraction = as.integer(100 * fraction / sum(fraction))) %>% 
  ungroup() %>% 
  print()
```

## Look at codon repeats and initiation mutations only
```{r, fig.width=3.6, fig.height=1.5}
plot_data <- count_data %>% 
  # look for a trinucleotide repeated 1 + 3 times
  filter(str_detect(codon, '([ACTG]{3})\\1{3}')) %>% 
  # extract codon
  mutate(codon_mutation = substr(codon, 1, 3)) %>% 
  # get the initiation mutation as either CTG at start or the -4 to -1 preceding ATG
  mutate(init_mutation = str_extract(init, '(?<=CAAA)CTG|[ACTG]{4}(?=ATG)')) %>% 
  # get only initiation mutations that we cloned 
  filter(init_mutation %in% names(initiationmutation_order)) %>% 
  # get only codon mutations that we cloned 
  filter(codon_mutation %in% names(codonmutation_order)) %>% 
  # arrange init_mutation in correct_order
  mutate(init_mutation = fct_reorder(
    init_mutation, initiationmutation_order[init_mutation])) %>% 
  # arrange codon_mutation in correct_order
  mutate(codon_mutation = fct_reorder(
    codon_mutation, codonmutation_order[codon_mutation])) %>% 
  # get rid of unwanted cols
  select(-barcode, -codon, -init) %>% 
  # spread cdna and gdna counts to adjacent columns
  spread(type, count) %>% 
  # calculate log2 fold change cdna / gdna
  mutate(lfc = log2(cdna)-log2(gdna)) %>% 
  mutate(codon_group = substr(codon_mutation, 1, 1)) %>% 
  # set the maximum lfc to be zero to see fold changes clearly
  group_by(codon_group, replicate) %>% 
  mutate(lfc = lfc - max(lfc)) %>% 
  ungroup() %>% 
  # average across 4 replicates
  group_by(codon_mutation, init_mutation, codon_group) %>% 
  summarize(avg_lfc = mean(lfc), std_lfc = sd(lfc) / sqrt(n())) %>% 
  ungroup() %>% 
  mutate(codon_mutation = if_else(codon_group == "A", 
                                  paste0("10×", codon_mutation),
                                  paste0("8×", codon_mutation))) %>% 
  mutate(codon_mutation = fct_relevel(codon_mutation, "10×AGA"))
                                  

plot_data %>% 
  # exclude the mutated start codon since this will be shown in previous panel
  filter(init_mutation != "CTG") %>% 
  ggplot(aes(x = init_mutation, y = avg_lfc, group = codon_mutation, shape = codon_mutation)) +
  facet_wrap(~codon_group, ncol = 2, scales = "free_y") +
  geom_line(color = "dimgrey") +
  geom_point(fill = "dimgrey") +
  # geom_errorbar(aes(ymin = avg_lfc - std_lfc, ymax = avg_lfc + std_lfc),
  #               width = 0.5) +
  scale_color_manual(values = cbPalette) +
  scale_shape_manual(values = c(21, 24, 22, 25)) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 6)) +
  scale_y_continuous(limits = c(-2.2, +0.2), breaks = -2:0) +
  labs(x = "Kozak (-4 to -1)", y = "mRNA level (log2, a.u.)", shape = "")
  

ggsave("../../../figures/mrna_stability_vs_initiation_rate_4codons_wt.pdf")
```