---
title: "Plot ribosome density around RQC stall in endogenous genes"
author: "`r Sys.info()[['user']]`"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: 
  github_document:
    toc: 2
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, message = F, warning = F)
```


# Load libraries
```{r}
library(GenomicAlignments)
library(GenomicFeatures)
library(Biostrings)
library(tidyverse)
library(plyranges)
library(biobroom)
library(rasilabRtemplates)
```

# Load genome and annotations
```{r}
genome <- BSgenome.Scerevisiae.UCSC.sacCer3::BSgenome.Scerevisiae.UCSC.sacCer3
annotations <- "/fh/fast/subramaniam_a/db/rasi/genomes/yeast/Saccharomyces_cerevisiae/sgd/S288C_reference_genome_R64-1-1_20110203/saccharomyces_cerevisiae_R64-1-1_20110208.gff" %>% 
  rtracklayer::readGFF() %>% 
  as_tibble()

tx <- annotations %>% 
  GRanges() %>% 
  filter(type == "CDS" & orf_classification == "Verified") %>% 
  select(Name) %>% 
  split(.$Name) %>% 
  print()
```

# Get TE from the the paper

```{r}
te_paper <- read_tsv("../annotations/GSE53313_Cerevisiae_RNA_RPF.txt", skip = 2) %>% 
  setNames(c("id", "total", "ribo", "te")) %>% 
  filter(!is.na(te) & ribo > 5 & total > 5) %>%
  print()
```

# Load RQC stalls for joining with high TE genes
```{r}
rqc_stalls <- read_tsv("../../rqc_stalls_in_yeast_orfs/tables/ngrams_annotated.tsv") %>% 
  mutate(stall = if_else(stall %in% c("KR", "P"), "KPR", stall)) %>%
  print()
```

# Find typical (median) location of stall for use in control

```{r}
median_stall_loc <- median(rqc_stalls[["pos"]]) %>% 
  print()
```

# Convert RQC stalls to genomic coordinates

```{r}
rqc_stalls_coords <- rqc_stalls %>% 
  mutate(seqname = id, start = pos*3 + 1) %>%
  mutate(end = start) %>%
  select(seqname, start, end, stall, id) %>%
  GRanges() %>%
  mapFromTranscripts(tx) %>% 
  # get rid of mitochondrial sequence
  filter(seqnames != "chrMito") %>% 
  mutate(id = rqc_stalls[xHits, "id"], stall = rqc_stalls[xHits, "stall"]) %>%
  select(-xHits, -transcriptsHits)

# check that the mapping was done correctly
rqc_stalls_coords %>% 
  anchor_5p() %>% 
  stretch(29) %>% 
  getSeq(genome, .) %>% 
  translate()
```

# Load the Ribo-seq alignments

We do not run the codecell below after the first time to save time.

```{r load_ribo_aln, eval=F}
aln <- readGAlignments("../processeddata/mono/accepted_hits.bam") %>% 
  print()
```


# Trim the Ribo-seq alignments to the P-site and calculate coverage separately for + and - strands

We do not run the codecell below after the first time to save time.

```{r ribo_cvg, eval=F}
cvg_plus <- aln[strand(aln) == "+"] %>% 
  qnarrow(start = 13, width = 1) %>% 
  coverage() %>% 
  print()

cvg_minus <- aln[strand(aln) == "-"] %>% 
  qnarrow(start = qwidth(.) - 12, width = 1) %>% 
  coverage() %>% 
  print()

rtracklayer::export.bw(cvg_plus, "../processeddata/mono/cvg_plus.bw")
rtracklayer::export.bw(cvg_minus, "../processeddata/mono/cvg_minus.bw")
```

# Load the RNA-seq alignments

We do not run the codecell below after the first time to save time.

```{r load_total_aln, eval=F}
aln <- readGAlignments("../processeddata/totalrz/accepted_hits.bam") %>% 
  print()
```

# Trim the RNA-seq alignments to the P-site and calculate coverage separately for + and - strands

We do not run the codecell below after the first time to save time.

```{r total_cvg, eval=F}
cvg_plus <- aln[strand(aln) == "+"] %>% 
  qnarrow(start = 13, width = 1) %>% 
  coverage() %>% 
  print()

cvg_minus <- aln[strand(aln) == "-"] %>% 
  qnarrow(start = qwidth(.) - 12, width = 1) %>% 
  coverage() %>% 
  print()

rtracklayer::export.bw(cvg_plus, "../processeddata/totalrz/cvg_plus.bw")
rtracklayer::export.bw(cvg_minus, "../processeddata/totalrz/cvg_minus.bw")
```

# Load pre-computed coverage

```{r load_cvg, eval=T}
ribo_cvg_plus <- rtracklayer::import.bw("../processeddata/mono/cvg_plus.bw") %>% 
  coverage(weight = "score")
ribo_cvg_minus <- rtracklayer::import.bw("../processeddata/mono/cvg_minus.bw") %>% 
  coverage(weight = "score")
total_cvg_plus <- rtracklayer::import.bw("../processeddata/totalrz/cvg_plus.bw") %>% 
  coverage(weight = "score")
total_cvg_minus <- rtracklayer::import.bw("../processeddata/totalrz/cvg_minus.bw") %>% 
  coverage(weight = "score")
```

```{r}
cvg <- c('ribo' = c(GRanges(ribo_cvg_plus, strand = "+"), GRanges(ribo_cvg_minus, strand = "-")),
  'total' = c(GRanges(total_cvg_plus, strand = "+"), GRanges(total_cvg_minus, strand = "-"))
  ) %>% 
  GRangesList() %>% 
  as_tibble() %>% 
  select(-group) %>% 
  rename(sample = group_name) %>% 
  filter(score > 0) %>% 
  GRanges()
```

```{r}
tx_counts <- mapToTranscripts(cvg, tx) %>% 
  GRanges() %>% 
  mutate(sample = cvg$sample[xHits], score = cvg$score[xHits]) %>% 
  as_tibble()
```

```{r}
te_calculated <- tx_counts %>% 
  group_by(seqnames, sample) %>% 
  summarize(counts = sum(width * score)) %>% 
  ungroup() %>% 
  spread(sample, counts) %>% 
  filter(ribo > 100 & total > 100) %>%
  mutate(te = log2(ribo / total / sum(ribo) * sum(total))) %>%
  dplyr::rename(id = seqnames) %>% 
  print()
```

# Plot calculated TE vs. TE pre-calculated by Weinberg et al.

```{r, fig.width=3, fig.height=3}
te_paper %>% 
  select(id, te) %>% 
  inner_join(te_calculated, by = "id") %>% 
  ggplot(aes(x = te.x, y = te.y)) +
  geom_point(alpha = 0.3) +
  scale_x_continuous(limits = c(-6.5, 2.5)) +
  scale_y_continuous(limits = c(-6.5, 2.5)) +
  geom_abline(intercept = 0, slope = 1)
```

# Look at high TE (paper) genes with potential RQC stalls 
```{r}
te_paper_stall_data <- te_paper %>% 
  left_join(rqc_stalls, by = "id") %>%
  arrange(desc(te)) %>%
  select(te, gene, pos, ngram, ngram_weight, everything()) %>%
  print()
```
  
# Plot TE from paper as a function of stall strength

```{r, fig.width=2, fig.height=2}
plot_data <- te_paper_stall_data %>% 
  mutate(ngram_weight = as.factor(if_else(is.na(ngram_weight), 0, 1))) %>% 
  group_by(ngram_weight) %>% 
  mutate(`n` = paste0("N = ", dplyr::n())) %>% 
  ungroup() %>% 
  mutate(ngram_weight = fct_recode(ngram_weight, `No stall` = "0", `Stall` = "1"))
  
plot_data %>% 
  ggplot(aes(x = ngram_weight, y = te, fill = ngram_weight)) +
  geom_violin() +
  geom_boxplot(width = 0.2, fill = "white", color = "black", outlier.shape = NA) +
  labs(x = "S. cerevisiae genes", y = "Translation efficiency (log2, a.u.)") +
  geom_text(aes(x = ngram_weight, label = n),
            data = plot_data %>% group_by(ngram_weight) %>% slice(1),
            y = -7, size = 2.8) +
  scale_y_continuous(limits = c(-7.2, 3)) +
  scale_fill_manual(values = cbPalette, guide = "none") +
  NULL

ggsave("../figures/distribution_of_te_paper_for_rqc_stall_containing_saccer_genes.pdf")
```

# Test if stall-containing genes have lower or higher TE from paper

```{r}
wilcox.test(te ~ ngram_weight, data = plot_data, alternative = "two.sided") %>% 
  tidy() %>% 
  gather() %>% 
  knitr::kable()
```

# Plot difference in TE between stall-containing and remaining genes

```{r}
plot_data %>% 
  group_by(ngram_weight) %>% 
  summarize(median(te)) %>% 
  knitr::kable()
```

# Look at high TE (calculated) genes with potential RQC stalls 

```{r}
te_calc_stall_data <- te_calculated %>% 
  left_join(rqc_stalls, by = "id") %>%
  arrange(desc(te)) %>%
  select(te, gene, pos, ngram, ngram_weight, everything()) %>%
  print()
```
  
# Plot calculated TE as a function of stall strength

```{r, fig.width=2, fig.height=2}
plot_data <- te_calc_stall_data %>% 
  mutate(ngram_weight = as.factor(if_else(is.na(ngram_weight), 0, 1))) %>% 
  group_by(ngram_weight) %>% 
  mutate(`n` = paste0("N = ", dplyr::n())) %>% 
  ungroup() %>% 
  mutate(ngram_weight = fct_recode(ngram_weight, `No stall` = "0", `Stall` = "1"))
  
plot_data %>% 
  ggplot(aes(x = ngram_weight, y = te, fill = ngram_weight)) +
  geom_violin() +
  geom_boxplot(width = 0.2, fill = "white", color = "black", outlier.shape = NA) +
  labs(x = "S. cerevisiae genes", y = "Translation efficiency (log2, a.u.)") +
  geom_text(aes(x = ngram_weight, label = n),
            data = plot_data %>% group_by(ngram_weight) %>% slice(1),
            y = -7, size = 2.8) +
  scale_y_continuous(limits = c(-7.2, 3)) +
  scale_fill_manual(values = cbPalette, guide = "none") +
  NULL

ggsave("../figures/distribution_of_te_calculated_for_rqc_stall_containing_saccer_genes.pdf")
```

# Test if stall-containing genes have lower or higher calculated TE 

```{r}
wilcox.test(te ~ ngram_weight, data = plot_data, alternative = "two.sided") %>% 
  tidy() %>% 
  gather() %>% 
  knitr::kable()
```

# Plot difference in TE between stall-containing and remaining genes

```{r}
plot_data %>% 
  group_by(ngram_weight) %>% 
  summarize(median(te)) %>% 
  knitr::kable()
```

# Calculate TE for only the region preceding each stall

```{r}
te_preceding_stalls <- tx_counts %>% 
  left_join(rqc_stalls %>% select(id, pos), by = c("seqnames" = "id")) %>% 
  mutate(pos = if_else(is.na(pos), median_stall_loc, pos)) %>% 
  filter(start < 3*pos + 1) %>% 
  group_by(seqnames, sample) %>% 
  summarize(counts = sum(width * score)) %>% 
  ungroup() %>% 
  spread(sample, counts) %>% 
  filter(ribo > 100 & total > 100) %>%
  mutate(te = log2(ribo / total / sum(ribo) * sum(total))) %>%
  dplyr::rename(id = seqnames) %>% 
  print()
```

# Look at high TE (calculated only preceding stall) genes with potential RQC stalls 

```{r}
te_preceding_stall_data <- te_preceding_stalls %>% 
  left_join(rqc_stalls, by = "id") %>%
  arrange(desc(te)) %>%
  select(te, gene, pos, ngram, ngram_weight, everything()) %>%
  print()
```
  
# Plot calculated TE for region preceding stalls as a function of stall strength

```{r, fig.width=2, fig.height=2}
plot_data <- te_preceding_stall_data %>% 
  mutate(ngram_weight = as.factor(if_else(is.na(ngram_weight), 0, 1))) %>% 
  group_by(ngram_weight) %>% 
  mutate(`n` = paste0("N = ", dplyr::n())) %>% 
  ungroup() %>% 
  mutate(ngram_weight = fct_recode(ngram_weight, `No stall` = "0", `Stall` = "1"))
  
plot_data %>% 
  ggplot(aes(x = ngram_weight, y = te, fill = ngram_weight)) +
  geom_violin() +
  geom_boxplot(width = 0.2, fill = "white", color = "black", outlier.shape = NA) +
  labs(x = "S. cerevisiae genes", y = "Translation efficiency (log2, a.u.)") +
  geom_text(aes(x = ngram_weight, label = n),
            data = plot_data %>% group_by(ngram_weight) %>% slice(1),
            y = -7, size = 2.8) +
  scale_y_continuous(limits = c(-7.2, 3)) +
  scale_fill_manual(values = cbPalette, guide = "none") +
  NULL

ggsave("../figures/distribution_of_te_preceding_stalls_for_rqc_stall_containing_saccer_genes.pdf")
```

# Test if stall-containing genes have lower or higher calculated TE preceding stall

```{r}
wilcox.test(te ~ ngram_weight, data = plot_data, alternative = "two.sided") %>% 
  tidy() %>% 
  gather() %>% 
  knitr::kable()
```

# Plot difference in TE between stall-containing and remaining genes

```{r}
plot_data %>% 
  group_by(ngram_weight) %>% 
  summarize(median(te)) %>% 
  knitr::kable()
```

# Session Info

```{r}
sessionInfo()
```

